articles/hapi-vs-koa-vs-express/index.md

```diff
@@ -0,0 +1,184 @@
+- [Introduction to Node.js](/engineering-education/history-of-nodejs/)
+
+### Framework histories
+Before we start our in-depth comparison of each framework, let's take a look at how they got started.
+
+#### Hapi.js
+Hapi.js was created by the mobile team at Walmart Labs — led by Eran Hammer to handle their traffic for events like Black Friday, since it's one of the busiest shoping days in the U.S. calendar
+.
+Hapi's initial commit was made on August 5th, 2011 and after 5,619 commits, it is currently at version 20.0.0. It was originally built using the Express framework before the challenges that drove Walmart to make Hapi, its stand-alone framework. 
+
+Refer to my [introduction to Hapi.js framework](/engineering-education/introduction-to-hapi/) for a detailed look at Hapi.js.
+
+#### Koa.js
+Koa's initial commit was made on August 17th, 2013 by TJ Holowaychuk. The framework was built by the team behind express with the aim of making it smaller, more expressive, and robust. Koa.js is currently at version 2.13.0 with 1,086 commits. 
+
+Koa.js is also known for having a smaller footprint compared to other frameworks. Refer to my [introduction to koa.js framework](/engineering-education/introduction-to-koajs/) for a detailed look at Koa.js.
+
+#### Express.js
+The Express framework is the oldest of the three. Its initial commit was made by TJ Holowaychuk on June 26th, 2009 and its first release was on the 2nd of January 2010 after around 660 commits. 
+
+The two main contributors at the time of its first release were TJ Holowaychuk and Ciaron Jessup. Express.js is currently at version 4.17.1 after 280 releases and 5,592 commits.
+
+### Framework popularity
+All these frameworks are widely used by developers worldwide, but still, one might need to know which framework is the most popular. Our popularity comparison will be based on statistics from GitHub and [npm](https://www.npmjs.com)
+
+#### Hapi.js
+Hapi.js's GitHub repository has 12.7k stars, 205 contributors, 1.3k forks and 332 releases.
+
+According to [npm](https://www.npmjs.com/package/hapi), 1,470 packages depend on Hapi.js.
+#### Koa.js
+Koa.js's GitHub repository has 29.9k stars, 218 contributors and 2.8k forks.
+
+According to [npm](https://www.npmjs.com/package/koa), 4,678 packages depend on Koa.js.
+#### Express.js
+Express.js's GitHub repository has 50k stars, 205 contributors, 8.3k forks and 280 releases.
+
+According to [npm](https://www.npmjs.com/package/express), 45,743 packages depend on Express.js.
+
+### Creating a server
+In this section we will have a look at how the frameworks differ in creating a server. Unlike Hapi, both Koa and Express use a similar logic to create a server.
+
+In Express, you first import the express module, then, instantiate the module and store it in a variable. Then use its listen method to start a server at a certain port.
+
+Let's see a code example to illustrate this.
+
+```JavaScript
```

Comment:
You should add that users need to `npm install express` first and the example can be run using `node (filename).js`. Also, add semicolons to the add of lines as a linting best practise.

Same for the other examples. Readers new to Node.js may not understand this.